src/discovery/Discovery.ts

```diff
@@ -123,36 +153,28 @@ class Discovery {
   } = {}): Promise<void> {
     this.logger.info(`Starting ${this.constructor.name}`);
     if (fresh) {
-      await this.db.clear(this.discoveryDbPath);
-    }
-    // Getting latest ID and creating ID generator
-    let latestId: DiscoveryQueueId | undefined;
-    const keyIterator = this.db.iterator(this.discoveryQueueDbPath, {
-      limit: 1,
-      reverse: true,
-      values: false,
-    });
-    for await (const [keyPath] of keyIterator) {
-      const key = keyPath[0] as Buffer;
-      latestId = IdInternal.fromBuffer<DiscoveryQueueId>(key);
+      // Cancel all tasks for discovery
+      for await (const task of this.taskManager.getTasks('asc', true, [
+        this.constructor.name,
+      ])) {
+        task.cancel(discoveryDestroyedTaskReason);
+      }
     }
-    this.discoveryQueueIdGenerator =
-      discoveryUtils.createDiscoveryQueueIdGenerator(latestId);
-    // Starting the queue
-    this.discoveryProcess = this.setupDiscoveryQueue();
+    this.taskManager.registerHandler(
+      this.discoverVertexHandlerId,
+      this.discoverVertexHandler,
+    );
     this.logger.info(`Started ${this.constructor.name}`);
   }
 
   public async stop(): Promise<void> {
     this.logger.info(`Stopping ${this.constructor.name}`);
-    this.queuePlug.resolveP();
-    await this.discoveryProcess;
+    this.taskManager.deregisterHandler(this.discoverVertexHandlerId);
     this.logger.info(`Stopped ${this.constructor.name}`);
   }
 
   public async destroy(): Promise<void> {
     this.logger.info(`Destroying ${this.constructor.name}`);
-    await this.db.clear(this.discoveryDbPath);
     this.logger.info(`Destroyed ${this.constructor.name}`);
   }
```

Comment:
`discoveryDestroyedTaskReason` is used to cancel a task when destroying. Currently it's only used when starting with `fresh: true`.  Doing it during `discovery.destroy` requires that the `TaskManager` is running when destroying. I haven't made this change yet and I'm not sure it's worth making it.

I'm leaning towards needing it.

Reply:
If we want to clear all the discovery tasks when destroying then we need `TaskManager` and `DB` to be running at the time. In `polykeyAgent.destroy` the `DB` is already started temporarily. So it's not a major change.

Reply:
`discovery.destroy()` is now cancelling any existing discovery tasks.